src/nbl/video/IGPUCommandBuffer.cpp

```diff
@@ -884,6 +884,30 @@ bool IGPUCommandBuffer::bindGraphicsPipeline(const IGPUGraphicsPipeline* const p
     return bindGraphicsPipeline_impl(pipeline);
 }
 
+bool IGPUCommandBuffer::bindRayTracingPipeline(const IGPURayTracingPipeline* const pipeline)
+{
+    // Because binding of the Gfx pipeline can happen outside of a Renderpass Scope,
+    // we cannot check renderpass-pipeline compatibility here.
+    // And checking before every drawcall would be performance suicide.
+    if (!checkStateBeforeRecording(queue_flags_t::COMPUTE_BIT))
+        return false;
+
+    if (!pipeline || !this->isCompatibleDevicewise(pipeline))
+    {
+        NBL_LOG_ERROR("incompatible pipeline device!");
+        return false;
+    }
+
```

Comment:
add checks for:
https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#VUID-vkCmdTraceRaysKHR-pRayGenShaderBindingTable-03681
https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#VUID-vkCmdTraceRaysKHR-pRayGenShaderBindingTable-03682
https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#VUID-vkCmdTraceRaysKHR-stride-03690

and friends